### ISSUE TITLE
Validate shouldn't apply to scripts below baseline on baselined schema

### ISSUE THREAD

**Issue by esauser** [flyway/flyway#2191]
###### Which version and edition of Flyway are you using?
5.2.1

###### If this is not the latest version, can you reproduce the issue with the latest one as well?
(Many bugs are fixed in newer releases and upgrading will often resolve the issue)
This issue does not occur on 5.1.4

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Command-line

###### Which database are you using (type & version)?
MSSQL 2014

###### Which operating system are you using?
Windows

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, the SQL statement that failed (if relevant) and the command you ran.)
flyway migrate

###### What did you expect to see?
Expected the operation to complete successfully, just as it does with 5.1.4

###### What did you see instead?
ERROR: Detected both transactional and non-transactional statements within same migration (even though mixed is false).

TL;DR we are using Flyway to manage an existing database so we baselineOnMigrate set to true.  We also have mixed set to false.  On the first run, this sets V1 as the baseline.  We then set V1__baseline.sql as the script to generate the baseline of the database locally.  In version 5.1.4 Flyway did not validate this file since it was never executed.  Now in version 5.2.1 it is validating the file, despite it never being executed.

You can easily reproduce this by creating your own project locally with baselineOnMigrate set to true, mixed set to false, and a V1 migration script that attempts to do something that cannot be done within a transaction.  On version 5.1.4 you'll see that the error doesn't occur while on version 5.2.1 it does occur.

I have reviewed the change log and nothing stood out to me as this being an intended change.

We are currently working around this by pinning to 5.1.4, but that won't work in perpetuity.  We could also get around it by moving the baseline script from V1_baseline.sql, but we prefer to leave it there since it makes sense.  We also have several pipelines in place that rely on this and developers working locally, all of which would be forced to take breaking changes.

**Comment by esauser** [comment]
![image](https://user-images.githubusercontent.com/24945592/47808506-84133f00-dd0c-11e8-9306-08f67c01f9b9.png)

![image](https://user-images.githubusercontent.com/24945592/47808579-a4db9480-dd0c-11e8-9fac-3bd2907163d7.png)

**Comment by esauser** [comment]
@axelfontaine why did this behavior change in a breaking way with `5.2.x`?

**Comment by axelfontaine** [comment]
@esauser Well let's say that the pre-5.2 behavior was more like an undocumented historical accident, which in fact already contradicted our docs for validate at the time. Things have been rectified now to ensure documentation and behavior match.
